test/k8s/res_limits_test.go

```diff
@@ -0,0 +1,512 @@
+					err = yaml.Unmarshal([]byte(out), &jsonEnvArray)
+					Expect(err).ToNot(HaveOccurred())
+					var idx = -1
+					for i, m := range jsonEnvArray {
+						if m["name"] == "GOMEMLIMIT" {
+							idx = i
+							break
+						}
+					}
+
+					op := "replace"
+					idxStr := fmt.Sprintf("%d", idx)
+					if idx == -1 {
+						op = "add"
+						idxStr = "-"
+					}
+
+					patchJson += fmt.Sprintf(`,{"op": "%s", "path": "/spec/template/spec/containers/0/env/%s", "value": {"name": "GOMEMLIMIT", "value":"%dMiB"}}`,
+						op, idxStr, memLimit)
+				}
+			}
+
+			patchJson += "]"
+			err := k8s.RunKubectlE(
+				cluster.GetTesting(),
+				cluster.GetKubectlOptions(Config.KumaNamespace),
+				"patch", "deployment", Config.KumaServiceName,
+				"--type=json",
+				"--patch", fmt.Sprintf(patchJson, miliCPU, memMega, miliCPU, memMega),
+			)
+			Expect(err).ToNot(HaveOccurred())
+
+			time.Sleep(2 * time.Second)
+
+			err = k8s.RunKubectlE(
+				cluster.GetTesting(),
+				cluster.GetKubectlOptions(Config.KumaNamespace),
+				"rollout", "status", "deployment", Config.KumaServiceName)
+			Expect(err).ToNot(HaveOccurred())
+		}
+
+		deployDPs := func() {
+			printLog("deploying %d services and %d instances per service\n", numServices, instancesPerService)
+
+			svcGraph := graph.GenerateRandomServiceMesh(872835240, numServices, 50, instancesPerService, instancesPerService)
+
+			buffer := bytes.Buffer{}
+			fakeServiceRegistry := "nicholasjackson"
+			if alternativeContainerRegistry != "" {
+				fakeServiceRegistry = alternativeContainerRegistry
+			}
+			Expect(svcGraph.ToYaml(&buffer, graph.ServiceConf{
+				WithReachableServices: true,
+				WithNamespace:         false,
+				WithMesh:              true,
+				Namespace:             TestNamespace,
+				Mesh:                  "default",
+				Image:                 fmt.Sprintf("%s/fake-service:v0.25.2", fakeServiceRegistry),
+			})).To(Succeed())
+
+			Expect(cluster.Install(YamlK8s(buffer.String()))).To(Succeed())
+		}
+
+		waitForDPs := func(ret chan<- bool) {
+			printLog("waiting for the data planes to become ready\n")
+			waitCh := make(chan bool)
+			go func(x chan<- bool) {
+				expectedNumOfPods := numServices * instancesPerService
+				created := Eventually(func() error {
+					opts := cluster.GetKubectlOptions(TestNamespace)
+					opts.Logger = logger.Discard
+					return k8s.WaitUntilNumPodsCreatedE(cluster.GetTesting(), opts,
+						metav1.ListOptions{}, expectedNumOfPods, 1, 0)
+				}, "10m", "3s").Should(Succeed())
+
+				pods, err := k8s.ListPodsE(cluster.GetTesting(), cluster.GetKubectlOptions(TestNamespace), metav1.ListOptions{})
+				if err != nil {
+					printLog("failed to list pods: %v\n", err)
+					x <- false
+					return
+				}
+
+				if created {
+					printLog("%d pods are now all created\n", expectedNumOfPods)
+				} else {
+					printLog("only %d of %d pods created\n", len(pods), expectedNumOfPods)
+					x <- created
+					return
+				}
```

Comment:
wouldn't it be easier to pass error channel here and do
```
if !created {
  errCh <- errors.Errorf("only %d of %d pods created", len(pods), expectedNumOfPods)
  return
}
Logf("%d pods are now all created\n", expectedNumOfPods)
```


at the end of the func just pass errCh<-nil?

Reply:
Yeah, it can be much simplified in this way.
Thanks.